c++/src/kj/compat/http.c++

```diff
@@ -1768,55 +1842,123 @@ kj::Own<kj::AsyncInputStream> HttpInputStreamImpl::getEntityBody(
     //   forwarding. We ignore the vague "ought to" part and implement the other two. (The
     //   dropping of Content-Length will happen naturally if/when the message is sent back out to
     //   the network.)
-    if (fastCaseCmp<'c','h','u','n','k','e','d'>(te->cStr())) {
+    if (fastCaseCmp<'c','h','u','n','k','e','d'>(te.cStr())) {
       // #3¶1
       return kj::heap<HttpChunkedEntityReader>(*this);
-    } else if (fastCaseCmp<'i','d','e','n','t','i','t','y'>(te->cStr())) {
+    } else if (fastCaseCmp<'i','d','e','n','t','i','t','y'>(te.cStr())) {
       // #3¶2
       KJ_REQUIRE(type != REQUEST, "request body cannot have Transfer-Encoding other than chunked");
       return kj::heap<HttpConnectionCloseEntityReader>(*this);
-    } else {
-      KJ_FAIL_REQUIRE("unknown transfer encoding", *te) { break; }
```

Comment:
Note that this `{ break; }` was likely important, since otherwise this exception (which is easily triggered by a remote party) would abort the process when exceptions are disabled. Another reason we should keep the code as it was.

Reply:
I was curious about that. I notice there's not a similar `{ break; }` on the `Content-Length` check below. Wouldn't that have the same issue potentially?